sci-mathematics/msieve/msieve-1.53.recipe

```diff
@@ -0,0 +1,66 @@
+SUMMARY="Library for Factoring Large Integers"
+DESCRIPTION="C library implementing a suite of algorithms to factor large integers"
+HOMEPAGE="https://sourceforge.net/projects/msieve/"
+COPYRIGHT="Public Domain"
+LICENSE="Public Domain"
+REVISION="1"
+SOURCE_URI="https://downloads.sourceforge.net/msieve/msieve/Msieve%20v$portVersion/msieve${portVersion/./}_src.tar.gz"
+CHECKSUM_SHA256="c5fcbaaff266a43aa8bca55239d5b087d3e3f138d1a95d75b776c04ce4d93bb4"
+PATCHES="msieve-1.53.patchset"
+
+ARCHITECTURES="?x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	msieve$secondaryArchSuffix = $portVersion
+	cmd:msieve$secondaryArchSuffix
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libecm$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	msieve${secondaryArchSuffix}_devel = $portVersion
+	devel:libmsieve$secondaryArchSuffix
+	"
+REQUIRES_devel="
+	haiku${secondaryArchSuffix}_devel
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libecm$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:sed
+	"
+
+PATCH()
```

Comment:
```diff
 	cmd:sed
 	"
 
+defineDebugInfoPackage msieve$secondaryArchSuffix \
+	$binDir/msieve
+
 PATCH()
 {
```

Reply:
Does `defineDebugInfoPackage` magically add `-g` to `CFLAGS`? If not then I don't think we have any debuginfo to extract. Moreover, it should be done for the lib, if that's even possible with static objects...

Reply:
I'm quite sure it does not add "-g".
But I've seen many packages which create the debuginfo package and which do not compile with "-g".
BTW, I'm still wondering if having debuginfo packages makes any sense if we don't compile with "-g".
Anyway, even if it does not make sense, at least we make the Haiku Debugger happy, by providing a debuginfo package.

Reply:
OK. If adding a debuginfo package does not make sense without the "-g", then let's forget about it.
No regrets ;-)

Reply:
Then I don't really agree with this. Those others packages should be reported and have "false" debuginfo removed. While Haiku Debugger might be happy, it's users won't. No one will expect a so-called "debuginfo" package to install, then figured out that they were merely placeholder.

I could add `-g` myself, but even so the debuginfo won't make sense, as `msieve` uses extensive inlining with `-O3`.
Compling `msieve` with less optimization is counter-intuitive, as the main point of science applications is to extract as much juice as possible from your hardware to solve a problem.

BTW, I think building package with debuginfo enabled should be a haikuporter feature, so that we could have optimized binaries for daily usage and upon request, debuginfo for development purposes